nupic/research/frameworks/dendrites/metrics/metrics.py

```diff
@@ -313,3 +314,126 @@ def entropy(x):
     _max_entropy = math.log(num_segments)
 
     return _entropy, _max_entropy
+
+
+def repr_overlap_matrix(activations, targets):
+    """
+    Returns a 2D torch tensor with shape (num_categories, num_categories) where cell
+    c1, c2 gives the mean value of pairwise representation overlap across all pairs of
+    examples between classes c1 and c2. Each individual pairwise representation overlap
+    is simply the fraction of hidden units that are active in both examples.
+
+    NOTE: This function is not specific to dendrites, and can be used with modules that
+    both include and don't include dendrite segments.
+
+    :param activations: 2D torch tensor with shape (batch_size, num_units) where entry
+                        b, i gives the activation of unit i for example b
+    :param targets: 1D torch tensor with shape (batch_size,) where entry b gives the
+                    target label for example b
+    """
+    with torch.no_grad():
+
+        repr_overlaps = repr_overlap_stats(activations, targets)
+
+        # Assume the following:
+        # - target values are zero-based
+        # - the largest target value in the batch is that amongst all data
+        num_categories = 1 + targets.max().item()
+
+        overlap_matrix = torch.zeros((num_categories, num_categories))
+
+        # For each pair of categories, compute the mean value of pairwise
+        # representation overlaps using all pairs of the form (x1, x2) where x1 belongs
+        # to category c1, and x2 belongs to category c2
+        for t1 in range(num_categories):
+            for t2 in range(1 + t1):
+
+                mean_overlap_val = np.mean(repr_overlaps[t1][t2])
+                overlap_matrix[t1, t2] = mean_overlap_val
+
+        return overlap_matrix
+
+
+def repr_overlap_distributions(activations, targets):
```

Comment:
You're not returning the distributions, but the values that are used to populate histogram bins. It may make sense to call it `representation_overlap_scores` or `representation_overlap_values`.

Reply:
Good point. Done.